homeassistant/components/kostal_plenticore/helper.py

```diff
@@ -183,9 +222,83 @@ async def _async_update_data(self) -> dict[str, dict[str, str]]:
         _LOGGER.debug("Fetching %s for %s", self.name, self._fetch)
 
         fetched_data = await client.get_setting_values(self._fetch)
+        return fetched_data
+
+
+class PlenticoreSelectUpdateCoordinator(DataUpdateCoordinator):
+    """Base implementation of DataUpdateCoordinator for Plenticore data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        name: str,
+        update_inverval: timedelta,
+        plenticore: Plenticore,
+    ) -> None:
+        """Create a new update coordinator for plenticore data."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name=name,
+            update_interval=update_inverval,
+        )
+        # data ids to poll
+        self._fetch = defaultdict(list)
+        self._plenticore = plenticore
+
+    def start_fetch_data(self, module_id: str, data_id: str, all_options: str) -> None:
+        """Start fetching the given data (module-id and entry-id)."""
+        self._fetch[module_id].append(data_id)
+        self._fetch[module_id].append(all_options)
+
+        # Force an update of all data. Multiple refresh calls
+        # are ignored by the debouncer.
+        async def force_refresh(event_time: datetime) -> None:
+            await self.async_request_refresh()
+
+        async_call_later(self.hass, 2, force_refresh)
+
+    def stop_fetch_data(self, module_id: str, data_id: str, all_options: str) -> None:
+        """Stop fetching the given data (module-id and entry-id)."""
+        self._fetch[module_id].remove(all_options)
+        self._fetch[module_id].remove(data_id)
+
+
+class SelectDataUpdateCoordinator(
+    PlenticoreSelectUpdateCoordinator, DataUpdateCoordinatorMixin
+):
+    """Implementation of PlenticoreUpdateCoordinator for select data."""
+
+    async def _async_update_data(self) -> dict[str, dict[str, str]]:
+        client = self._plenticore.client
+
+        if client is None:
+            return {}
+
+        _LOGGER.debug("Fetching select %s for %s", self.name, self._fetch)
+
+        fetched_data = await self.async_get_currentoption(self._fetch)
 
         return fetched_data
 
+    async def async_get_currentoption(
```

Comment:
Please separate words with underscore in variable and function names.

Reply:
Fixed in https://github.com/home-assistant/core/pull/59998